src/main/java/com/vincit/faculdade/CalculadoraDeMediaDeAlunos.java

```diff
@@ -0,0 +1,38 @@
+package com.vincit.faculdade;
+
+import java.util.List;
+
+public class CalculadoraDeMediaDeAlunos {
+
+    public double calcularMedia(List<Aluno> list) {
+        double somaDeNotas = 0;
+        int contador = 0;
+
+        for (Aluno aluno : list) {
+            double nota = aluno.notas().get(contador);
```

Comment:
Bem interessante,
neste ponto voce esta buscando a nota de acordo com o "contato"
our seja do aluno1 usa a nota da posicao 0, o aluno2 da posicao1, o aluno3 na posicao2 e assim em diante, podendo causar um resultado inesperado... (falha minha na escrita do cenario ;) )

Atualizei este test que deve te ajudar e ajustar esse ponto :)

```java
    @Test
    void calcularMediaDeveCalcularApenasAlunosComNotasValidasQuandoPossuiAlunoComNotaInvalida() {
        var mediaFinalEsperada = 7;
        var notasInvalidas = List.of(-1, 10, 5);
        var notaAluno1 = List.of(10, 10, 10);
        var notaAluno2 = List.of(10, 0, 5);
        var notaAluno3 = List.of(4, 9, 5);
        var alunos = List.of(
                new Aluno(notasInvalidas),
                new Aluno(notaAluno1),
                new Aluno(notaAluno2),
                new Aluno(notaAluno3)
        );

        var mediaFinalDosAlunos = calculadoraMedia.calcularMedia(alunos);

        assertThat(mediaFinalDosAlunos).isEqualTo(mediaFinalEsperada);
    }
```

Reply:
Tem razão, me deixei levar pelo teste positivo e não percebi o problema das iterações utilizando o contador, corrigi o problema adicionando um novo iterador for para percorrer as notas dos alunos, não sei se era a melhor abordagem mas funcionou legal, o novo teste realmente me deu uma boa visão do problema, obrigado.